Source install: make instructions for disabling moveit packages prominent (#277)
ros-planning/moveit.ros.org: install/source/index.markdown

* Ubuntu 15.10, 16.04 / [ROS Kinetic](http://www.ros.org/wiki/kinetic/Installation/Ubuntu)
* Ubuntu 14.04 / [ROS Indigo](http://www.ros.org/wiki/indigo/Installation/Ubuntu)

> As of February 2019, [MoveIt for ROS 2.0](https://github.com/ros-planning/moveit2) development has begun but is not ready yet.

Reviewer: Seems like an unrelated change?

Author: This PR updated the source install page. This is part of the source install page, and I don't think is very controversial or complex. So I see no harm in improving the documentation here, also.

## Prerequisites

### Install ROS

Follow all the instructions to [install ROS](http://wiki.ros.org/ROS/Installation). Please make sure you have followed all steps and have the latest versions of packages installed:

    rosdep update
    sudo apt-get update
    sudo apt-get dist-upgrade

Source installation requires [wstool](http://wiki.ros.org/wstool), [catkin_tools](https://catkin-tools.readthedocs.io/en/latest/), and optionally [clang-format](https://clang.llvm.org/docs/ClangFormat.html):

    sudo apt-get install python-wstool python-catkin-tools clang-format-3.9

### Create Workspace and Source

Optionally create a new workspace, you can name it whatever:

    mkdir ~/ws_moveit
    cd ~/ws_moveit

Next, source your ROS workspace to load the necessary environment variables, depending on what version of ROS you installed.

Choose one of the three:

    source /opt/ros/melodic/setup.bash
    source /opt/ros/kinetic/setup.bash
    source /opt/ros/indigo/setup.bash

This will load the ``${ROS_DISTRO}`` variable, needed for the next step.

## Download Source Code

> By default, we will assume you are building on the latest branch -  *master*.

Reviewer: ```suggestion
By default, we will assume you are building the latest branch -  `master`.
```

> This branch builds for ROS Kinetic and newer, e.g. Ubuntu 16.04 and newer.

Reviewer: ```suggestion
This branch builds for ROS Kinetic and newer, e.g. on Ubuntu 16.04 and newer.
```

> If you would like to build an older release of MoveIt! from source, see the section below.

Reviewer: Which section?

Author: The ones below

    wstool update -t src
    rosdep install -y --from-paths src --ignore-src --rosdistro ${ROS_DISTRO}
    catkin config --extend /opt/ros/${ROS_DISTRO} --cmake-args -DCMAKE_BUILD_TYPE=Release

> ## Optional: Disable Packages from Building

Reviewer: ```suggestion
## Optional: Excluding Packages from a Build
```

> MoveIt is a large project and the defailt compile time can take ~30 minutes.

Reviewer: ```suggestion
MoveIt is a large project and the default compile time can easily take around 30 minutes.
```

> If you would like to reduce your compile time, you can disable certain unneeded packages from being built.

Reviewer: ```suggestion
If you would like to reduce your compile time or only need to build a subset of all the packages, `catkin` allows you to configure only a subset of packages to include in a build.
```

> Be careful not to disable packages that are needed dependency for others.

Reviewer: ```suggestion
Be careful not to disable packages that are needed by others (ie: are a dependency of other packages).
```

> There are several approaches to disabling packages:

Reviewer: ```suggestion
You can exclude packages from a build in several ways:
```

Author: You're splitting hairs here with this rewording... disabling vs excluding... they mean the same thing here

Reviewer: Just ignore it if you don't agree.

It's a suggestion.

 - ``catkin-tools --blacklist`` (recommended)

>  - ``touch CATKIN_IGNORE``

Reviewer: ```suggestion
 - ``touch /path/to/package/directory/CATKIN_IGNORE``
```

>  - ``rm -rf *package_name*``

Reviewer: ```suggestion
 - ``rm -rf /path/to/package/directory``
```

> Here we show some examples using ``catkin-tools``. Note: If you have already built these packages in your workspace you will also need to use ``catkin clean``.

Reviewer: ```suggestion
Here we show some examples using ``catkin_tools``. Note: If you have already built these packages in your workspace you will need to use ``catkin clean`` first.
```

### Disable All High-Level User Interfaces (optional)

    catkin config --blacklist \
         moveit_commander \
         moveit_setup_assistant \
         moveit_ros_robot_interaction \
         moveit_ros_visualization \
         moveit_ros_benchmarks \
         moveit_controller_manager_example \
         moveit_ros_warehouse \
         moveit_ros_manipulation \
         moveit_visual_tools \
         rviz_visual_tools

### Disable CHOMP Motion Planer (optional)

> Note every command to ``catkin config --blacklist`` will empty the previous list.

Reviewer 2: ```suggestion
Note, every command to ``catkin config --blacklist`` will override the previous list by default. Using the option ``--append-args`` you can augment the existing blacklist.
```

---

## Advanced

### Building Older Releases Of MoveIt!

> Its best to contribute to our latest branch, even if you're still on an earlier version of Ubuntu. However our latest branch does not support older versions of ROS such as ROS Indigo. Use the following command to build from source older releases:

Reviewer: ```suggestion
It's best to contribute to our latest branch, even if you're still on an earlier version of Ubuntu. However our latest branch does not support older versions of ROS such as ROS Indigo. Use the following command to build older releases from source:
```

>     catkin config --extend /opt/ros/${ROS_DISTRO} --cmake-args -DCMAKE_BUILD_TYPE=Release

Reviewer: ```suggestion
    catkin config --extend /opt/ros/${ROS_DISTRO} --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

Author: We've always advised people to build for release. I'm not convinced this is a good change (what if they really are just deploying this?). Please suggest this change in a separate PR.

Reviewer: I reviewed this with a developer in mind. If you're not going to be working on MoveIt, why build from source? The first thing I have to ask people when they post issues about crashing code or other strange things is to build everything with debug symbols enabled.

Why not preempt that sort of situation and have them use `RelWithDebInfo`.

But I'm completely fine with keeping `Release`.
